File: crud-item-list/src/test/java/com/gurunars/crud_item_list/CheckerEditTest.java
package com.gurunars.crud_item_list;

import org.junit.Test;
import org.mockito.internal.util.collections.Sets;

import java.util.Arrays;
import java.util.HashSet;
import java.util.List;

import java8.util.function.Consumer;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class CheckerEditTest {

    private final ActionEdit<String> checkerEdit = new ActionEdit<>(new Consumer<String>() {
        @Override
        public void accept(String s) {

        }
    });
    private final List<String> all = Arrays.asList("one", "two");

    @Test
    public void selectionOfOne_leadsToTrue() throws Exception {
        assertTrue(checkerEdit.canPerform(all, Sets.newSet("one")));
    }

    @Test
    public void noSelection_leadsToFalse() throws Exception {
        assertFalse(checkerEdit.canPerform(all, new HashSet<String>()));
    }

    @Test
    public void selectionOfMultipl_leadsToFalse() throws Exception {
        assertFalse(checkerEdit.canPerform(all, Sets.newSet("one", "two")));
    }

}

File: crud-item-list/src/test/kotlin/com/gurunars/crud_item_list/CheckerEditTest.kt
package com.gurunars.crud_item_list

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class CheckerEditTest {

    private val checkerEdit = ActionEdit<StringItem>({})
    private val all = listOf("one", "two")

    @Test
    @Throws(Exception::class)
    fun selectionOfOne_leadsToTrue() {
        checkerEdit.canPerform(all, setOf("one"), { assertTrue(it) })
    }

    @Test
    @Throws(Exception::class)
    fun noSelection_leadsToFalse() {
        checkerEdit.canPerform(all, setOf(), { assertFalse(it) })
    }

    @Test
    @Throws(Exception::class)
    fun selectionOfMultipl_leadsToFalse() {
        checkerEdit.canPerform(all, setOf("one", "two"), { assertFalse(it) })
    }
}